File: code/1431.kids-with-the-greatest-number-of-candies.php
<?php
/*
 * @lc app=leetcode id=1431 lang=php
 *
 * [1431] Kids With the Greatest Number of Candies
 */

// @lc code=start
class Solution {

    /**
     * @param Integer[] $candies
     * @param Integer $extraCandies
     * @return Boolean[]
     */
    function kidsWithCandies($candies, $extraCandies) {
        // my solution
        $max = max($candies);
        $result = [];
        foreach ($candies as $candy) {
            $result[] = $candy + $extraCandies >= $max;
        }
        return $result;

        // e.g.
        // $max = max($candies);
        // return array_map(fn($candy) => $candy + $extraCandies >= $max, $candies);
    }
}
// @lc code=end

File: code/1431.kids-with-the-greatest-number-of-candies.go
package code

/*
 * @lc app=leetcode id=1431 lang=golang
 *
 * [1431] Kids With the Greatest Number of Candies
 */

// @lc code=start
func kidsWithCandies(candies []int, extraCandies int) []bool {
	answer := make([]bool, 0, len(candies))

	max := 0
	for _, v := range candies {
		if v > max {
			max = v
		}
	}

	for _, v := range candies {
		answer = append(answer, v+extraCandies >= max)
	}

	return answer
}

// @lc code=end
